components/nimbus/src/stateless/cirrus_client.rs

```diff
@@ -134,7 +134,7 @@ impl CirrusClient {
         // part of https://mozilla-hub.atlassian.net/browse/EXP-3401
         let nimbus_id = Uuid::new_v4();
         let available_randomization_units =
-            AvailableRandomizationUnits::with_client_id(client_id.as_str());
+            AvailableRandomizationUnits::with_user_id(user_id.as_str());
```

Comment:
Correct! 

> since it is a part of the overall schema it should not be behind conditional compilation.

^ this comment I made in the description was in reference to a particular configuration syntax Rust has where we can conditionally compile lines of code behind crate features.

```rust
pub enum RandomizationUnit {
    NimbusId,
    ClientId,
    #[cfg(not(feature = "stateful"))]
    UserId,
}
```

In the above example, the `UserId` enum value would not exist in the resulting binary if the `stateful` feature is enabled when compiling the `nimbus-sdk` library.

Since `UserId` is going to be a part of the core experimentation schema, I'm advocating for it to remain in all binaries to prevent accidental where deserializing JSON could result in errors if an app gets an experiment with the wrong randomization unit.

Reply:
Awesome! 💯